### ISSUE TITLE
Gruds in Space - .WOZ Boot Error

### ISSUE THREAD

**Issue by GreatHierophant** [AppleWin/AppleWin#921]
Gruds in Space [in](https://archive.org/details/wozaday_Gruds_in_Space) the A Woz A Day collection fails to boot in AppleWin 1.29.16.0.  Applesauce reports no errors when I converted the .a2rs in the archive to .woz myself, but the game will eventually report a boot error after showing alternating blue and orange lines.  On my (unenhanced) 128K Apple //e the same .woz image will boot with a wDrive and I can play the game.  

More information about what this protection is doing can be found [here](https://archive.org/details/GrudsInSpace4amCrack).

**Comment by tomcw** [comment]
Thanks, I can reproduce this "BOOT ERROR".

Reproduced on II+, //e and enhanced //e models, using AppleWin 1.29.16.0 (dev build).

**Comment by tomcw** [comment]
Aside, running as an Apple II+ model from the debugger with 'brk on' resulted in some AppleWin debugger crashes. These have been fixed in f3b6b86.

**Comment by tomcw** [comment]
Triage/development note:

1. In DataLatchReadWriteWOZ(), with significantBitCells = 100: `BOOT ERROR`
2. Comment out this: (and with significantBitCells = 100): boots OK
```
	if (i == bitCellRemainder-1)			// On last bitCell
		m_latchDelay += extraLatchDelay;	// +0 or +2
```
3. Comment out same code as 2 (reverting to significantBitCells = 50): boots OK

NB. the code in 2, is the alt fix to Wasteland here: https://github.com/AppleWin/AppleWin/issues/733#issuecomment-576026713

But reverting to before this "alt fix to Wasteland" (ie. the original Wasteland fix) doesn't work either:
```
	UINT extraLatchDelay = (UINT)floppy.m_extraCycles ? 1 : 0;	// GH#733
	:
	m_latchDelay += extraLatchDelay;
	extraLatchDelay = 0;
```

**Comment by tomcw** [comment]
The protection is on track 14 (T$0E): [gruds-track14.txt](https://github.com/AppleWin/AppleWin/files/6014942/gruds-track14.txt)

At a high-level, there's a nibble-counting function at $6908:  (called from $68BF)

```
6908	; entrypoint
	$6bd8 = 0(h), $6bd9 = 0(l)	; nibble count
6918	; first C0EC access after seeking to T$0E (track-14 / phase 28)

. read 50 ($32) nibbles	; read enough 9-bit nibbles to sync after seek (or on retry)
. read nibbles until read #$DD (NB. there is just *one* $DD nibble on the track)
. read 8 nibbles	; skip the 4x 4&4 nibbles + 2x nibbles 2x 9-bit syncs
. loop:
	if read nibble == #$DD then goto $6964
	else inc 6bd9, and inc 6bd8 on overflow; then loop

6964: read 4&4 (AF,BF) = $1F, store to: $6bdd
6979: read 4&4 (AB,BE) = $16, store to: $6bdc
```

Then after calling the above function:
```
pass protection if: (6bd8).w == (6bdc).w == $161F
```

**Comment by tomcw** [comment]
Basically Gruds in Space isn't working because of the extra delay fix for Wasteland + LOA.

**Comment by tomcw** [comment]
Fixed by #930 / b5726aa.
